app/src/main/java/com/example/myapplication/ThirdFragment.java

```diff
@@ -36,51 +30,122 @@ public class ThirdFragment extends Fragment {
 
 
     private FragmentThirdBinding binding;
-
+    ViewGroup v = null;
+    private static ColorStateList purple = ColorStateList.valueOf(Color.parseColor("#6750A3"));
+    private static ColorStateList blue = ColorStateList.valueOf(Color.parseColor("#73C2F0"));
 
     @Override
     public View onCreateView(
             LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState
     ){
         binding = FragmentThirdBinding.inflate(inflater, container, false);
+        v = container;
+            setSwitchColor(binding.switch1);
+            setSwitchColor(binding.switch2);
+            setSwitchColor(binding.switch3);
+        binding.switch2.setChecked(MainActivity.defense);
+        binding.switch3.setChecked(MainActivity.ground);
+        binding.switch3.setThumbTintList(blue);
+        binding.switch3.setTrackTintList(blue);
+
         return binding.getRoot();
     }
 
+    private void setSwitchColor(Switch switch1) {
+        if (switch1.isChecked()) {
+            switch1.setThumbTintList(purple);
+            switch1.setTrackTintList(purple);
+        } else {
+            switch1.setThumbTintList(blue);
+            switch1.setTrackTintList(blue);
+        }
+    }
+
     public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+
         binding.toAutoCont.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
                 NavHostFragment.findNavController(ThirdFragment.this)
                         .navigate(R.id.action_ThirdFragment_to_thirdFragment2);
             }
         });
+        ObjectAnimator animation = ObjectAnimator.ofFloat(binding.pop, "rotationX", new float[]{0f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f});
+        animation.setDuration(1000);
+        binding.pop.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                animation.start();
+                MainActivity.darkMode = !MainActivity.darkMode;
+                UIHelpers.lightDark(v, MainActivity.darkMode);
+
+            }
+        });
+        binding.switch1.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
```

Comment:
these methods are all changing MainActivity.broke, should they be changing different values instad?

Reply:
deleted broke